app/controllers/AssessmentController.scala

```diff
@@ -71,20 +72,29 @@ class AssessmentController @Inject() (
         val hasAssessmentBeenAnswered =
           request.userAnswers.get(AssessmentPage(recordId, index)).exists(_ != NotAnsweredYet)
 
-        if (exemptions.isEmpty) {
-          Future.successful(
-            Redirect(
-              navigator.nextPage(AssessmentPage(recordId, index, shouldRedirectToCya = true), mode, request.userAnswers)
-            )
-          )
-        } else if (areWeRecategorising && hasAssessmentBeenAnswered && mode == NormalMode) {
-          Future.successful(
-            Redirect(
-              navigator.nextPage(AssessmentPage(recordId, index), mode, request.userAnswers)
+        val shouldDisplayNext = areWeRecategorising && hasAssessmentBeenAnswered && mode == NormalMode
+
+        val commodityCodeWithoutZeros = commodityCode.reverse.dropWhile(char => char == '0').reverse
+
+        val shouldGoToLongerCommodityCode =
+          exemptions.isEmpty && category == 2 && commodityCodeWithoutZeros.length <= 6 &&
+            categorisationInfo.descendantCount != 0
+
+        val shouldGoToCya = exemptions.isEmpty && !shouldGoToLongerCommodityCode
+
+        (shouldGoToCya, shouldGoToLongerCommodityCode, shouldDisplayNext) match {
```

Comment:
If this looks less than optimal, please consider it is very likely to be refactored or replaced in the coming sprint.

Reply:
Yes it definitely is not going to make the cut 😆

Reply:
Exactly :')